blocks/library/quote/index.js

```diff
@@ -59,9 +64,23 @@ registerBlockType( 'core/quote', {
 			{
 				type: 'block',
 				blocks: [ 'core/text' ],
-				transform: ( { value, citation } ) => {
+				transform: ( { value, citation, ...attrs } ) => {
+					const textElement = Array.isArray( value ) ? value[ 0 ] : value;
```

Comment:
You were right, it seems like the ternary is not necessary here. I only needed a defaultValue for newly inserted quotes.